sympy/polys/matrices/tests/test_domainmatrix.py

```diff
@@ -538,3 +539,25 @@ def test_DomainMatrix_hstack():
     B = DomainMatrix([[QQ(3), QQ(4)], [QQ(5), QQ(6)]], (2, 2), QQ)
     AB = DomainMatrix([[QQ(1), QQ(3), QQ(4)], [QQ(2), QQ(5), QQ(6)]], (2, 3), QQ)
     assert A.hstack(B) == AB
+
+
+def test_DomainMatrix_scalarmul():
+    A = DomainMatrix([[ZZ(1), ZZ(2)], [ZZ(3), ZZ(4)]], (2, 2), ZZ)
+    lamda = DomainScalar(QQ(3)/QQ(2), QQ)
+    assert A * lamda == DomainMatrix([[QQ(3)/QQ(2), QQ(3)], [QQ(9)/QQ(2), QQ(6)]], (2, 2), QQ)
+    assert A * 2 == DomainMatrix([[ZZ(2), ZZ(4)], [ZZ(6), ZZ(8)]], (2, 2), ZZ)
+    assert A * DomainScalar(ZZ(0), ZZ) == DomainMatrix([[ZZ(0)]*2]*2, (2, 2), ZZ)
+    assert A * DomainScalar(ZZ(1), ZZ) == A
+
+    raises(TypeError, lambda: A * 1.5)
+
+
+def test_DomainMatrix_truediv():
+    A = DomainMatrix.from_Matrix(Matrix([[1, 2], [3, 4]]))
+    lamda = DomainScalar(QQ(3)/QQ(2), QQ)
+    assert A / lamda == DomainMatrix([[QQ(2)/QQ(3), QQ(4)/QQ(3)], [QQ(2), QQ(8)/QQ(3)]], (2, 2), QQ)
+    assert A / 2 == DomainMatrix([[RR(0.5), RR(1.0)], [RR(1.5), RR(2.0)]], (2, 2), RR)
```

Comment:
If the domain of `A` is `ZZ` then dividing like this should lead to `QQ`.

Reply:
Yes, but in `DomainMatrix.__truediv__` seems to construct `1/ lamda.element` in `RR` domain.

```py
In [14]: lamda = DomainScalar(ZZ(2), ZZ)

In [15]: a=DomainScalar.from_sympy(1/lamda.element)

In [16]: a.domain
Out[16]: RR
```

Kindly share your opinion.
Thanks.

Reply:
This is resolved by the above discussion.
Thanks.